cmd/kubeadm/app/util/staticpod/utils.go

```diff
@@ -233,7 +233,7 @@ func ReadStaticPodFromDisk(manifestPath string) (*v1.Pod, error) {
 		return &v1.Pod{}, errors.Wrapf(err, "failed to read manifest for %q", manifestPath)
 	}
 
-	obj, err := kubeadmutil.UnmarshalFromYaml(buf, v1.SchemeGroupVersion)
+	obj, err := kubeadmutil.UnmarshalFromYaml(buf)
 	if err != nil {
 		return &v1.Pod{}, errors.Errorf("failed to unmarshal manifest for %q from YAML: %v", manifestPath, err)
 	}
```

Comment:
below there is a `pod := obj.(*v1.Pod)`
it can panic if the user passes a file that's not a v1.Pod manfiest file.

it can be changed to:
```
pod, ok := obj.(*v1.Pod)
if !ok {
	return &v1.Pod{}, errors.Errorf("failed to parse Pod object defined in %q", manifestPath)
}
```

Reply:
Updated.